content/articles/todolist-with-svelte-and-firebase/index.md

```diff
@@ -0,0 +1,1135 @@
+          {/each}
+        </div>
+      </div>
+    </div>
+  ```
+
+  We've updated the button to call the markTodo function and pass the right arguments.
+
+- `header.svelte`
+  
+  ```js
+  <script>
+    import { logOut } from "./../services/firebase";
+    import { UserStore } from "./../stores";
+  </script>
+
+  <header>
+    {#if $UserStore}
+      <img src={$UserStore.photoURL} alt="User" />
+      <p>Welcome, {$UserStore.displayName.split(" ")[0]}</p>
+    {/if}
+    <button on:click={logOut}> Log Out </button>
+  </header>
+  ```
+
+  The 'Log Out' button can now log users out.
+
+- `edit-todo.svelte`
+  
+  ```js
+    <script>
+      import { UserStore } from "./../stores";
+      import { addTodo } from "./../services/firebase";
+    
+      let name = "";
+      let priority = "1";
+    
+      const submitTodo = () => {
+        const newTodo = {
+          priority,
+          name,
+          completed: false,
+        };
+        name = ''
+        priority = '1'
+        addTodo(newTodo, $UserStore.uid);
+      };
+    
+    </script>
+    
+    <form>
+      <header>Add New Todo</header>
+    
+      <label for="name">
+        Todo Name
+        <input bind:value={name} id="name" type="text" />
+      </label>
+    
+      <label for="priority">
+        Todo Priority
+        <input
+          bind:value={priority}
+          id="priority"
+          min="1"
+          step="1"
+          max="5"
+          type="range"
+        />
+      </label>
+    
+      <button on:click|preventDefault={submitTodo}>Add Todo</button>
+    </form>
+  ```
+
+  This updates the button so it can call the addTodo function now.
+
+- `App.svelte`
+  
+  ```js
+    <script>
+      import router from "page";
+      import Home from "./pages/home.svelte";
+      import Login from "./pages/login.svelte";
+      import { getAuth, onAuthStateChanged } from "firebase/auth";
+      import { UserStore } from "./stores";
+      import { onMount } from "svelte";
+      import { getTodos } from "./services/firebase";
+
+      const auth = getAuth();
+      onMount(async () => {
+        onAuthStateChanged(auth, async (user) => {
+          if (user) {
+            UserStore.set(user);
+            getTodos(user.uid);
+          } else {
+            UserStore.set(user);
+            getTodos(user.uid);
+            console.log("User signed out");
+          }
+        });
+      });
+
+      let page;
+
+      router("/", (ctx, next) => (page = Home));
+
+      router("/login", (ctx, next) => (page = Login));
+
+      router.start();
+    </script>
+
+    <svelte:component this={page} />
+
+  ```
+
+We call the `onMount` function to help us run a function after a component has been mounted. The `onAuthStateChanged` function calls its callback whenever the user state changes. We then update the `UserStore` based on the value we receive.
+  
+Clear the stores in the `stores.js` file so Firebase can take over and !Voila! we're done.
```

Comment:
Make sure to give a proper ending to the tutorial with demo images or GIFs. Don't just end it abruptly.

Reply:
Make sure to explain how to run.